pkg/runner/retriever/retriever.go

```diff
@@ -52,13 +52,14 @@ func GetMCPServer(
 	imageManager := images.NewImageManager(ctx)
 	// Check if the serverOrImage is a protocol scheme, e.g., uvx://, npx://, or go://
 	if runner.IsImageProtocolScheme(serverOrImage) {
+		logger.Debugf("Attempting to retrieve MCP server from protocol scheme: %s", serverOrImage)
 		var err error
 		imageToUse, imageMetadata, err = handleProtocolScheme(ctx, serverOrImage, rawCACertPath, imageManager)
 		if err != nil {
 			return "", nil, err
 		}
 	} else {
-		logger.Debugf("No protocol scheme detected, using image: %s", serverOrImage)
+		logger.Debugf("No protocol scheme detected, attempting to retrieve remote server or registry server: %s", serverOrImage)
```

Comment:
If it reaches this point it's either an image (`thv run ghcr.io/example/mcp-server:latest`) or a registry server (`thv run fetch`). The registry server may or may not be remote, but the important designation is that it's being looked up in the registry.

```suggestion
		logger.Debugf("No protocol scheme detected, attempting to retrieve image or registry server: %s", serverOrImage)
```